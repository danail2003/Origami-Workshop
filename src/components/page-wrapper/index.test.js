import React from 'react';
import renderer from 'react-test-renderer';
import PageWrapper from './index';
import UserContext from '../../context';
import { Route, BrowserRouter } from 'react-router-dom';

describe('PageWrapper', () => {
    it('should render authenticated routes', () => {
        const component = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        user: {
                            loggedIn: true,
                            id: '456'
                        }
                    }}>
                        <PageWrapper />
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>
        ).toJSON();
        expect(component).toMatchSnapshot()
    });

    it('should render non-authenticated routes', () => {
        const component = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        user: {
                            loggedIn: false
                        }
                    }}>
                        <PageWrapper />
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>
        ).toJSON();
        expect(component).toMatchSnapshot()
    });
});