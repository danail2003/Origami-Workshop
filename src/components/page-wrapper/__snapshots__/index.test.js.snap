// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageWrapper should render authenticated routes 1`] = `
<div
  className="app"
>
  <nav
    className="navigation"
  >
    <ul>
      <img
        alt="origami"
        className="logo"
        src="white-origami-bird.png"
      />
      <li
        className="header-list-item"
      >
        <a
          className="header-link"
          href="/"
          onClick={[Function]}
        >
          Publications
        </a>
      </li>
      <li
        className="header-list-item"
      >
        <a
          className="header-link"
          href="/share"
          onClick={[Function]}
        >
          Share your thoughts
        </a>
      </li>
      <li
        className="header-list-item"
      >
        <a
          className="header-link"
          href="/profile/456"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="container"
  >
    <aside
      className="aside"
    >
      <ul>
        <li
          className="aside-list-item"
        >
          <a
            className="aside-link"
            href="/"
            onClick={[Function]}
          >
            Publications
          </a>
        </li>
        <li
          className="aside-list-item"
        >
          <a
            className="aside-link"
            href="/share"
            onClick={[Function]}
          >
            Share your thoughts
          </a>
        </li>
        <li
          className="aside-list-item"
        >
          <a
            className="aside-link"
            href="/profile/456"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
      </ul>
    </aside>
    <main
      className="main"
    />
  </div>
  <footer
    className="footer"
  >
    <ul>
      <li
        className="footer-list-item"
      >
        <a
          className="footer-link"
          href="/"
          onClick={[Function]}
        >
          Publications
        </a>
      </li>
      <li
        className="footer-list-item"
      >
        <a
          className="footer-link"
          href="/share"
          onClick={[Function]}
        >
          Share your thoughts
        </a>
      </li>
      <li
        className="footer-list-item"
      >
        <a
          className="footer-link"
          href="/profile/456"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
      <img
        alt="origami"
        className="logo"
        src="blue-origami-bird-flipped.png"
      />
    </ul>
    <p
      className="paragraph"
    >
      Software University © 2019
    </p>
  </footer>
</div>
`;

exports[`PageWrapper should render non-authenticated routes 1`] = `
<div
  className="app"
>
  <nav
    className="navigation"
  >
    <ul>
      <img
        alt="origami"
        className="logo"
        src="white-origami-bird.png"
      />
      <li
        className="header-list-item"
      >
        <a
          className="header-link"
          href="/"
          onClick={[Function]}
        >
          Publications
        </a>
      </li>
      <li
        className="header-list-item"
      >
        <a
          className="header-link"
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
      <li
        className="header-list-item"
      >
        <a
          className="header-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="container"
  >
    <aside
      className="aside"
    >
      <ul>
        <li
          className="aside-list-item"
        >
          <a
            className="aside-link"
            href="/"
            onClick={[Function]}
          >
            Publications
          </a>
        </li>
        <li
          className="aside-list-item"
        >
          <a
            className="aside-link"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </li>
        <li
          className="aside-list-item"
        >
          <a
            className="aside-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </aside>
    <main
      className="main"
    />
  </div>
  <footer
    className="footer"
  >
    <ul>
      <li
        className="footer-list-item"
      >
        <a
          className="footer-link"
          href="/"
          onClick={[Function]}
        >
          Publications
        </a>
      </li>
      <li
        className="footer-list-item"
      >
        <a
          className="footer-link"
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
      <li
        className="footer-list-item"
      >
        <a
          className="footer-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      <img
        alt="origami"
        className="logo"
        src="blue-origami-bird-flipped.png"
      />
    </ul>
    <p
      className="paragraph"
    >
      Software University © 2019
    </p>
  </footer>
</div>
`;
